lib/internal/test_runner/test.js

```diff
@@ -448,5 +464,23 @@ class Test extends AsyncResource {
     }
   }
 }
+class Suite extends Test {
+  constructor(options) {
+    super(options);
+
+    this.runInAsyncScope(this.fn);
+    this.fn = () => {};
+  }
+  async run() {
+    this.parent.activeSubtests++;
+    this.startTime = hrtime();
+    const subtests = this.skipped ? [] : this.subtests;
+    for (const subtest of subtests) {
+      await subtest.run();
+    }
```

Comment:
```suggestion
    ArrayPrototypeReduce(subtests, (prev, subtest) => {
      await prev;
      await subtest.run();
    }, Promise.resolve());
```
since for-of uses Symbol.iterator on Array.prototype

Reply:
Let’s not forget to `await` the resulting promise. 
```suggestion
    await ArrayPrototypeReduce(subtests, async (prev, subtest) => {
      await prev;
      await subtest.run();
    }, PromiseResolve());
```